File: ios/Sources/App/AppConfig.swift
import Foundation

/// App-level constants that are not clinical content. The GitHub repo slug lives
/// here (one place) so the "Flag as outdated" links match the web client.
/// Mirror any change in web/src/lib/appConfig.js.
enum AppConfig {
    static let githubRepo = "awpeace1906-collab/kairos"

    /// "0.1.0" — read from the bundle (MARKETING_VERSION in project.yml) so the
    /// Settings/About screen never needs a manual edit when the version bumps.
    static var appVersion: String {
        (Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String) ?? "?"
    }

    /// Prefilled "content outdated" GitHub issue URL for a module's metadata.
    static func flagOutdatedURL(_ meta: RecordMeta) -> URL? {
        let appVersion = (Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String) ?? "?"
        let build = (Bundle.main.infoDictionary?["CFBundleVersion"] as? String) ?? "?"
        let tier: String
        switch meta.reviewTier {
        case .level(let n): tier = String(n)
        case .stable: tier = "stable"
        case nil: tier = "—"
        }
        let title = "Outdated: \(meta.title) (\(meta.id) v\(meta.contentVersion))"
        let body = """
        **Module:** \(meta.id) (\(meta.contentType.rawValue))
        **Version:** v\(meta.contentVersion)
        **Last verified:** \(meta.lastReviewed ?? "—")
        **Next review due:** \(meta.nextReviewDue ?? "—")
        **Review tier:** \(tier)
        **Client:** iOS \(appVersion) (\(build))

        **What's outdated / newer source:**


        """
        var c = URLComponents(string: "https://github.com/\(githubRepo)/issues/new")
        c?.queryItems = [
            URLQueryItem(name: "labels", value: "content,needs-review"),
            URLQueryItem(name: "title", value: title),
            URLQueryItem(name: "body", value: body),
        ]
        return c?.url
    }

    /// General "report an issue" link for the Settings screen — no module context,
    /// just points at the repo's issue tracker.
    static func newIssueURL(title: String = "", body: String = "") -> URL? {
        var c = URLComponents(string: "https://github.com/\(githubRepo)/issues/new")
        var items: [URLQueryItem] = []
        if !title.isEmpty { items.append(URLQueryItem(name: "title", value: title)) }
        if !body.isEmpty { items.append(URLQueryItem(name: "body", value: body)) }
        c?.queryItems = items.isEmpty ? nil : items
        return c?.url
    }
}

File: web/src/lib/appConfig.js
// App-level constants that are not clinical content. Keep the GitHub repo slug
// here (one place) so the "Flag as outdated" links in both clients point at the
// same issue tracker. Mirror any change in ios/Sources/App/AppConfig.swift.

export const GITHUB_REPO = "awpeace1906-collab/kairos";

/** "0.1.0" — the single place to bump the web client's version. Keep in sync
    with MARKETING_VERSION in ios/project.yml (the two platforms build/ship
    independently, so this can't be derived automatically the way the iOS
    About screen reads it from its own bundle — bump both when releasing). */
export const APP_VERSION = "0.1.0";

/** Build a prefilled "content outdated" GitHub issue URL for a module's meta. */
export function flagOutdatedURL(mod) {
  const title = `Outdated: ${mod.title} (${mod.id} v${mod.content_version ?? "?"})`;
  const body = [
    `**Module:** ${mod.id} (${mod.contentType ?? "?"})`,
    `**Version:** v${mod.content_version ?? "?"}`,
    `**Last verified:** ${mod.last_reviewed ?? "—"}`,
    `**Next review due:** ${mod.next_review_due ?? "—"}`,
    `**Review tier:** ${mod.review_tier ?? "—"}`,
    `**Client:** web · ${navigator.userAgent}`,
    ``,
    `**What's outdated / newer source:**`,
    ``,
  ].join("\n");
  const q = new URLSearchParams({ labels: "content,needs-review", title, body });
  return `https://github.com/${GITHUB_REPO}/issues/new?${q.toString()}`;
}

/** General "report an issue" link for the Settings screen — no module context. */
export function newIssueURL() {
  return `https://github.com/${GITHUB_REPO}/issues/new`;
}
